### DISCUSSION TITLE
Groovy incremental compilation

### DISCUSSION THREAD

**Discussion by tcrespog** [micronaut-projects/micronaut-core#8241] [Q&A] [answered]
Can Micronaut Groovy projects take advantage of incremental compilation?

I've built this sample multiproject made of a Groovy subproject and a Java subproject: https://github.com/tcrespog/mn-incremental.
In the Groovy subproject, I've enabled incremental compilation by following the Gradle guidelines regarding Groovy [compilation avoidance]( https://docs.gradle.org/current/userguide/groovy_plugin.html#sec:groovy_compilation_avoidance) and [incremental compilation]( https://docs.gradle.org/current/userguide/groovy_plugin.html#sec:incremental_groovy_compilation); other than that, the Java subproject is a dependency of the Groovy subproject. 

As far as I can see, compilation avoidance works: a change (ABI-compatible) in the Java subproject does not trigger a recompilation of the Groovy project.

However, I'd say incremental compilation is not working: the `compileGroovy` task execution time values do not vary greatly when comparing compilations with incremental compilation enabled vs disabled. Nevertheless, in the Java subproject, I can notice significant differences in elapsed time and lines in the output such as: `Note: Creating bean classes for 2 type elements`.

**Comment by jeffscottbrown** [comment]
@melix Do you know the answer to the question above?

**Comment by melix** [comment, marked as answer]
Adding `-i` to the build gives us the following:

> Full recompilation is required because Incremental annotation processing is not supported by Groovy

So we have our answer: incremental compilation doesn't work with Groovy in the case of Micronaut, because of annotation processing.
